src/sinks/util/buffer/compression.rs

```diff
@@ -99,10 +99,7 @@ impl<'de> de::Deserialize<'de> for Compression {
                             if level.is_some() {
                                 return Err(de::Error::duplicate_field("level"));
                             }
-                            level = Some(match map.next_value::<usize>() {
-                                Ok(value) => value.to_string(),
-                                Err(_) => map.next_value::<&str>()?.to_string(),
-                            });
+                            level = Some(map.next_value::<&str>()?);
```

Comment:
Is this right? Will it automatically convert `usize`s into `&str`s?

Reply:
Moved to owned PR: https://github.com/timberio/vector/pull/5429

@jszwedko thank you for a question! I added tests and realized that there new bug :facepalm: